argocd/schema_application.go

```diff
@@ -1166,6 +1166,383 @@ func applicationSpecSchemaV2() *schema.Schema {
 	}
 }
 
+func applicationSpecSchemaV3() *schema.Schema {
```

Comment:
This new schema does not seem to be used anywhere, unless I'm mistaken?

Reply:
Oops. Yes, this isn't used. I should have updated the following as well https://github.com/oboukili/terraform-provider-argocd/blob/24699dfe067dd4116512c1b7bb6f7158ffbb476d/argocd/resource_argocd_application.go#L30

For now I have removed this but it may be worth reviewing this decision in light of the comment/discussion below.